app/models/manageiq/providers/hawkular/middleware_manager/refresh_parser.rb

```diff
@@ -71,7 +71,7 @@ def parse_datasource(server, datasource, config)
           :nativeid          => datasource.id,
           :ems_ref           => datasource.path
         }
-        unless config.empty? || config['value'].empty?
+        unless config.empty? || config['value'].empty? || !config['value'].is_a?(Hash)
           data[:properties] = config['value'].select { |k, _| k != 'Username' && k != 'Password' }
```

Comment:
You could use [Hash.except](http://api.rubyonrails.org/classes/Hash.html#method-i-except)

Also, if your concern is not to leak passwords, I'd also check for `:Password, :password`